Added custom button menu to Resource Details page

chalettu opened the pull request:
@miq-bot add_label  gaprindashvili/yes
@miq-bot add_label enhancement
BZ - https://bugzilla.redhat.com/show_bug.cgi?id=1513113

chalettu commented:
Before SS 
![before_ss](https://user-images.githubusercontent.com/230549/32960046-ca57f9a4-cb80-11e7-859c-0e0fef9263d3.gif)

chalettu commented:
After SS
![new_menu](https://user-images.githubusercontent.com/230549/32960054-d493db4a-cb80-11e7-9c4e-821dce802960.gif)

Loicavenel commented:
@AllenBW I am confused here. the design if I am not mistaking is to have the kebab menu when there are too many Custom button on the top bar not all the time...

Loicavenel commented:
@serenamarie125 need your comment here..

chalettu commented:
@Loicavenel , no need to ask @AllenBW , I was the one working on this issue.  This kebab menu will not show all the time, we put a limit of 3 custom buttons or button groups to show on the screen before we show the kebab menu.

Loicavenel commented:
@chalettu my apologise..

chalettu commented:
No worries @Loicavenel .  Screenshots and small videos can sometimes make it hard to convey other logic behind what is intended.

chriskacerguis commented:
@chalettu could you please address the CC issues?

@AllenBW I think what @chalettu is showing is that it is limiting the buttons now (it's not all the time)...at least that is not what the desired behavior is.  @chalettu could you clarify?

chalettu commented:
@serenamarie125 , I agree that button spacing is an issue.  Chris K had created another issue to address it for clarity sake and being consistent.  
SS is just an abbreviation for screen shot.  So the screenshots are for demo purposes.  The code as implemented will only default to kebab menu after 3 buttons/button groups are defined.  We can change this count if you like.  
The tooltips are aren't necessary and we can remove them if you feel they don't fit for that drop down.  Please let me know and we can definitely have them removed.

serenamarie125 commented:
LOL - i'm being too literal sorry @chalettu now i get the "SS"

tooltips are fine, @Loicavenel just showed me that you can set hover text, so that makes sense.

I do see an issue that the drop down menu has a caret at the top ... it should be straight, similar to the user menu shown below
![image](https://user-images.githubusercontent.com/12504403/33043559-e16554e4-ce13-11e7-8693-73a03cf700a3.png)

chalettu commented:
@serenamarie125 , Here is a SS with the caret removed.  Please let me know if we think any more changes might be required.  
![testing](https://user-images.githubusercontent.com/230549/33045408-e8bc5d64-ce09-11e7-817a-4faa2b4ae0f8.gif)

serenamarie125 commented:
@chalettu this is really more complex :\

Kebab uses caret (http://www.patternfly.org/pattern-library/widgets/#kebabs)
- note that the kebab has a selected state and the caret should be centered - this screenshot is from the code example from the link above
<img width="174" alt="screen shot 2017-11-20 at 8 13 23 pm" src="https://user-images.githubusercontent.com/12504403/33049721-6114206a-ce2f-11e7-8589-f8d1607a17ce.png">


The regular menus ( snapshots & power operations) shouldn't

AllenBW commented:
@chalettu Any update on centering of the kebab dropdown ?

chalettu commented:
@AllenBW , haven't come back to this one.  I will try and take a look at centering the kebab.

AllenBW commented:
@chalettu I suspect this might be a less is more kinda situation... rip out classes see what changes... 🤔

chalettu commented:
Updated SS where I added back the caret and centered it. 
![updated_caret](https://user-images.githubusercontent.com/230549/33683352-d9c9e42c-da87-11e7-958f-bfdeed64235d.gif)

chalettu commented:
@serenamarie125 , can you take a look at this latest screenshot and see if it matches your expectations of this caret menu?

serenamarie125 commented:
@miq-bot add_labels ux/approved

serenamarie125 commented:
@miq-bot remove_labels ux/review

serenamarie125 commented:
@miq-bot add_label ux/approved

serenamarie125 commented:
@miq-bot remove_label ux/review

AllenBW commented on client/app/services/custom-button-menu/custom-button-menu.component.js:
```diff
@@ -0,0 +1,62 @@
+      }
+    }
+
+    function postFailure () {
```
There's gonna be real valuable server feedback on the error here, [something similar to this](https://github.com/manageiq/manageiq-ui-service/blob/01c16e684a29a8aba449249b8028c79f223eead3/client/app/services/process-snapshots-modal/process-snapshots-modal.component.js) should be used here

chalettu commented:
@AllenBW , I updated based on feedback. Let me know if you approve.

simaishi commented:
Gaprindashvili backport details:

```
$ git log -1
commit 0b8e5730075e7feead9d824be159191f2124679a
Author: Allen Wight <allen.b.wight@gmail.com>
Date:   Fri Dec 8 07:10:47 2017 -0500

    Merge pull request #1254 from chalettu/custom-button-grouping
    
    Added custom button menu to Resource Details page
    (cherry picked from commit 672d1281341c8e7f818d1779682679e9727abfa4)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1524718
```
